developer/reference/articles/front-end/03-front-end-taglibs/30-liferay-portlet-objects-intro.markdown

```diff
@@ -0,0 +1,33 @@
+---
+header-id: liferay-portlet-objects-available-in-jsps
+---
+
+# Liferay Portlet Objects Available in JSPs
```

Comment:
@shuyangzhou Can you (or a team member) review this reference on the objects available to JSPs via `<portlet:defineObjects>`?

Reply:
Hi @jhinkey 
we have a new phase from portlet 3.0 : HEADER_PHASE, so we added `headerRequest` and `headerResponse` in defineObjects. Also we have `liferayPortletRequest` and `liferayPortletResponse` defined no matter which phase you are in.
Could you please add these?

Thanks.
Tina.

Reply:
@tinatian - I'll do that. Thanks, Tina!